### PR TITLE
edit_line improvements

### PR DESCRIPTION
**Pull request by CalvinWalzel**
- Added support for self-closing tags to `Scarpe::HTML` (called `void_tags`)
- Refactored `edit_line` to use `Scarpe::HTML`
- Implemented `Scarpe::Dimensions` in `button` & `edit_line`
- Added width option to `edit_line`
- Added default value option to `edit_line`
- Ignored `.DS_Store` files in repo

### CODE REVIEW

File: lib/scarpe/button.rb [hunk 1]
```diff
@@ -36,11 +36,11 @@ def render
     def style
       styles = {}
 
-      styles[:width] = "#{@width}px" if @width
-      styles[:height] = "#{@height}px" if @height
+      styles[:width] = ::Scarpe::Dimensions.length(@width) if @width
```
Comment (TobiasBales) [inline lib/scarpe/button.rb:39]: this could be `Dimensions.length`, correct?

File: lib/scarpe/edit_line.rb [hunk 2]
```diff
@@ -33,7 +34,22 @@ def render
...
+        h.input(id: object_id, oninput: oninput, value: @text, style: style)
+      end
+    end
+
+    private
+
+    def style
+      styles = {}
+
+      styles[:width] = ::Scarpe::Dimensions.length(@width) if @width
```
Comment (TobiasBales) [inline lib/scarpe/edit_line.rb:50]: same here

File: test/test_html.rb [hunk 3]
```diff
@@ -37,4 +37,10 @@ def test_works_with_multiple_children
 
     assert_equal "<ul><li>one</li><li>two</li></ul>", subject
   end
+
+  def test_works_with_void_tag
+    subject = ::Scarpe::HTML.render { |h| h.input(type: "text") }
```
Comment (TobiasBales) [inline test/test_html.rb:42]: I think I copied this over here too, `HTML` would probably good enough

**Review by TobiasBales — CHANGES_REQUESTED** [review]
small nit, rest looks great!

File: test/test_html.rb (hunk 3 above)
Comment (TobiasBales) [reply test/test_html.rb:42]: well actually `Scarpe::HTML` in this scope I think

File: lib/scarpe/html.rb [hunk 4]
```diff
@@ -1,6 +1,8 @@
 module Scarpe
   class HTML
-    TAGS = %i[div p button ul li]
+    CONTENT_TAGS = %i[div p button ul li].freeze
+    VOID_TAGS = %i[input].freeze
```
Comment (Schwad) [inline lib/scarpe/html.rb:4]: Freebie for someone else; if you were rewriting image, you could put it in here too etc. etc.

File: test/test_html.rb (hunk 3 above)
Comment (CalvinWalzel) [reply test/test_html.rb:42]: In the tests, this doesn't work: `NameError: uninitialized constant TestHTML::HTML`

**Review by TobiasBales — APPROVED** [review]

**Pull request merged**